src/Resources/app/administration/src/module/findologic-module/components/findologic-config/index.js

```diff
@@ -1,196 +1,215 @@
-
-                let comparison = 0;
-                if (case1 > case2) {
-                    comparison = order === 'asc' ? 1 : -1;
-                } else if (case1 < case2) {
-                    comparison = order === 'asc' ? -1 : 1;
-                }
-                return comparison;
-            };
-        },
-
-        /**
-         * @public
-         */
-        openSalesChannelUrl() {
-            if (this.selectedSalesChannelId !== null) {
-                const criteria = new Criteria();
-                criteria.addFilter(
-                    Criteria.equals('id', this.selectedSalesChannelId)
-                );
-                criteria.setLimit(1);
-                criteria.addAssociation('domains');
-                this.salesChannelRepository.search(criteria, Shopware.Context.api).then((searchresult) => {
-                    const domain = searchresult.first().domains.first();
-                    this._openStagingUrl(domain);
-                });
-            } else {
-                this._openDefaultUrl();
-            }
-        },
-
-        /**
-         * @private
-         */
-        _openDefaultUrl() {
-            const url = `${window.location.origin}?findologic=on`;
-            window.open(url, '_blank');
-        },
-
-        /**
-         * @param {Object} domain
-         * @private
-         */
-        _openStagingUrl(domain) {
-            if (domain) {
-                const url = `${domain.url}?findologic=on`;
-                window.open(url, '_blank');
-            } else {
-                this._openDefaultUrl();
-            }
-        },
-
-        /**
-         * @public
-         */
-        getCategories() {
-            const criteria = new Criteria(1, 500);
-
-            if (this.term) {
-                criteria.setTerm(this.term);
-            }
-
-            this.isLoading = true;
-            const categoryRepository = this.repositoryFactory.create('category');
-
-            const translatedCategories = [];
-            categoryRepository.search(criteria, Shopware.Context.api).then((items) => {
-                this.total = items.total;
-                items.forEach((category) => {
-                    translatedCategories.push({
-                        value: category.id,
-                        label: category.translated.breadcrumb.join(' > ')
-                    });
-                });
-
-                this.categories = translatedCategories.sort(this.compare('label'));
-            }).finally(() => {
-                this.isLoading = false;
-            });
+    /**
+     * @public
+     * @param value
+     * @returns {boolean}
+     */
+    checkBoolFieldInheritance (value) {
+      return typeof value !== 'boolean'
+    },
+
+    /**
+     * @public
+     * @param prop
+     * @param order
+     * @returns {function(*, *): number}
+     */
+    compare (prop = 'name', order = 'asc') {
```

Comment:
`compare` is a weird name, especially when you have an optional parameter called `order`. Actually the order parameter hides the fact that you actually just compare weither something is bigger or smaller than the other value. That method is also pretty much useless on it's own, as it is strictly tight to the `sort` method of any search result entity.

I would suggest rewriting that method to accept any repository search result, and orders them by the given order parameter. This way you do not need to handle a callback or manually need to put the callback into the sort function of the search result.